magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleExtensions.kt

```diff
@@ -2,58 +2,58 @@ package com.wealthfront.magellan.lifecycle
 
 import kotlin.reflect.KProperty
 
-public fun <T : LifecycleAware, R> LifecycleOwner.lifecycle(lifecycleAware: T, getter: (T) -> R): Lifecycle<T, R> {
-  return Lifecycle(
+public fun <Child : LifecycleAware, Property> LifecycleOwner.attachFieldToLifecycle(lifecycleAware: Child, getter: (Child) -> Property): LifecycleChild<Child, Property> {
```

Comment:
nit: not even sure whether to mention this, but how do you feel about `ChildType` and `PropertyType`, to make it more clear that they're generic? I defer to you though, I've looked at this code too much haha.